force-app/main/default/lwc/edaSettingsTable/edaSettingsTable.js

```diff
@@ -0,0 +1,22 @@
+import { LightningElement, api } from "lwc";
+//custom labels
+import stgBtnAddMapping from "@salesforce/label/c.stgBtnAddMapping";
+
+export default class EdaSettingsTable extends LightningElement {
+    @api tabletitle;
+    @api tabledescription;
+    @api tabledata;
+    @api tablecolumns;
+
+    labelReference = {
+        newButtonLabel: stgBtnAddMapping,
+    };
+
+    handleNewRowClick(event) {
```

Comment:
If we are going to make this generic, then let's add in some events that we can use later.

Reply:
Should we add them in edit/delete/new items

Reply:
If we are using a genericized table, then we should add genericized events as well. This can be a simple passthrough that replicates the important details. My understanding is that handleNewRowClick on its own doesn't bubble and is captured at the component level, so we should create a new custom event and dispatch it so we can handle it in the parent component.

The same would apply to the row action event. Replicate the relevant data and pass it upstream.